Author: Alessio Treglia <quadrispro@ubuntu.com>
Description:
 Don't rely on xterm, try to use the user's preferred terminal emulator first.
 The option to change the window title differ between terminal emulators, try
 to set the right one.
Forwarded: no
---
 Runtime/codelite_xterm |   17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

--- codelite.orig/Runtime/codelite_xterm
+++ codelite/Runtime/codelite_xterm
@@ -9,11 +9,24 @@ if [ $# -ne 2 ]; then
 fi
 
 program_title=$1
+terminal_emulator=$(basename $(readlink -e /usr/bin/x-terminal-emulator))
+
+case ${terminal_emulator} in
+	"gnome-terminal")
+		title_opt="-t"
+		;;
+	*)
+		title_opt="-T"
+		;;
+esac
+
+echo $title_opt
+
 if [ "${LD_LIBRARY_PATH}" = "" ]; then
 	## LD_LIBRARY_PATH is not defined OR empty
 	## Run xterm without the bash wrapper
-	xterm -T "$program_title" -e $2 2> /dev/null
+	x-terminal-emulator "${title_opt}" "${program_title}" -e $2 2> /dev/null
 else
-	xterm -T "$program_title" -e /bin/bash -c 'export LD_LIBRARY_PATH=$0;shift;$@' $LD_LIBRARY_PATH $@ 2> /dev/null
+	x-terminal-emulator "${title_opt}" "${program_title}" -e /bin/bash -c 'export LD_LIBRARY_PATH=$0;shift;$@' $LD_LIBRARY_PATH $@ 2> /dev/null
 fi
 
